app/api/routers/token.py

```diff
@@ -645,9 +657,73 @@ def search_transfer_histories(
         stmt = stmt.where(IDXTransfer.source_event == data.source_event.value)
     if data.data is not None:
         stmt = stmt.where(cast(IDXTransfer.data, String).like("%" + data.data + "%"))
+    if data.transaction_hash is not None:
+        stmt = stmt.where(
+            IDXTransfer.transaction_hash.like("%" + data.transaction_hash + "%")
+        )
+    if data.created_from is not None:
+        stmt = stmt.where(
+            IDXTransfer.created >= data.created_from.astimezone(timezone.utc)
+        )
+    if data.created_to is not None:
+        stmt = stmt.where(
+            IDXTransfer.created <= data.created_to.astimezone(timezone.utc)
+        )
+    if data.value is not None and data.value_operator is not None:
+        match data.value_operator:
+            case ValueOperator.EQUAL:
+                stmt = stmt.where(IDXTransfer.value == data.value)
+            case ValueOperator.GTE:
+                stmt = stmt.where(IDXTransfer.value >= data.value)
+            case ValueOperator.LTE:
+                stmt = stmt.where(IDXTransfer.value <= data.value)
 
     count = session.scalar(select(func.count()).select_from(stmt.subquery()))
 
+    def _order(_order):
+        if _order == 0:
+            return asc
+        else:
+            return desc
+
+    if (
+        data.sort_item == "from_account_address_list"
+        and len(data.account_address_list) > 0
+    ):
+        stmt = stmt.order_by(
+            _order(data.sort_order)(
+                case(
+                    {
+                        account_address: i
+                        for i, account_address in enumerate(data.account_address_list)
+                    },
+                    value=IDXTransfer.from_address,
+                )
+            )
+        )
+    elif (
+        data.sort_item == "to_account_address_list"
+        and len(data.account_address_list) > 0
+    ):
+        stmt = stmt.order_by(
+            _order(data.sort_order)(
+                case(
+                    {
+                        account_address: i
+                        for i, account_address in enumerate(data.account_address_list)
+                    },
+                    value=IDXTransfer.to_address,
+                )
+            )
+        )
```

Comment:
Added `from_account_address_list` `to_account_address_list` to sort_item list so that the result can be ordered in arbitary order.